### PR TITLE
Split deployment related artifacts from 'example' to a new module named 'deploy'

### PR DESCRIPTION
**Pull request by kaniyan**
Deployment related artifacts which were part of 'example' are moved to a new module 'deploy'.

### CODE REVIEW

File: deploy/pom.xml [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                            <tarLongFileMode>posix</tarLongFileMode>
+                            <finalName>eclipse-hono-deploy-${project.version}</finalName>
+                            <appendAssemblyId>false</appendAssemblyId>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (sophokles73) [inline deploy/pom.xml]: missing EOL

**Comment by sophokles73** [comment]
can you also add a few lines to the release notes describing this change?

File: deploy/pom.xml (hunk 1 above)
Comment (kaniyan) [reply deploy/pom.xml]: Added

**Comment by kaniyan** [comment]
I have also updated the release notes with the description of this change.

File: site/content/release-notes.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ title = "Release Notes"
 
 * The Device Registry's AMQP endpoints can now be configured with the number of credits they should flow to clients connecting to the endpoints. The default value is 100. See [Device Registry admin guide]({{< relref "admin-guide/device-registry-config.md#service-configuration" >}}) for details.
 * Hono's service components can now be configured to use OpenSSL instead of the JVM's default SSL engine. The [admin guide]({{< relref "admin-guide/secure_communication.md#using-openssl" >}}) describes how to do this.
+* Hono deployment scripts are available under new module named "deploy". These scripts which were under module "example" were moved to "deploy".
```
Comment (sophokles73) [inline site/content/release-notes.md]: we do not use double quotes to emphasize. please use single back quotes in order to mark as literals.

Comment (kaniyan) [reply site/content/release-notes.md]: I have updated the description accordingly.

**Comment by sophokles73** [comment]
@kaniyan can you rebase this PR?

**Comment by sophokles73** [comment]
@kaniyan I wanted you to rebase your PR, not include everything from master ...

**Pull request merged**